src/communication_abstraction/src/client_handle_helper.c

```diff
@@ -0,0 +1,273 @@
+/**
+ * @file client_handle_helper.c 
+ * @brief Implements an abstract interface for communicating through the ModuleClient or DeviceClient libaries.
+ *  
+ * @copyright Copyright (c) 2019, Microsoft Corp.
+ */
+
+#include "aduc/client_handle_helper.h"
+
+#include <azureiot/iothub_device_client_ll.h>
+#include <azureiot/iothub_module_client_ll.h>
+
+static ADUC_ConnType g_ClientHandleType = ADUC_ConnType_NotSet;
+
+/**
+ * Note: Keeping incase there is a need for some static function nonsense
+ */
+_Bool ClientHandle_SetClientHandleType(ADUC_ConnType type)
+{
+    if (type != ADUC_ConnType_NotSet && g_ClientHandleType != ADUC_ConnType_NotSet)
+    {
+        return false;
+    }
+
+    g_ClientHandleType = type;
+
+    return true;
+}
+
+IOTHUB_DEVICE_CLIENT_LL_HANDLE GetDeviceClientHandle(ADUC_ClientHandle handle)
+{
+    return *(IOTHUB_DEVICE_CLIENT_LL_HANDLE*)handle;
+}
+
+IOTHUB_MODULE_CLIENT_LL_HANDLE GetModuleClientHandle(ADUC_ClientHandle handle)
+{
+    return *(IOTHUB_MODULE_CLIENT_LL_HANDLE*)handle;
+}
+
+/**
+ * 
+ */
```

Comment:
not the best function level comment I've ever seen :-)

Reply:
It's an invisible ink. Need to light a fire.... :)

---
In reply to: [591977401](https://github.com/Azure/iot-hub-device-update/pull/12#discussion_r591977401) [](ancestors = 591977401)

Reply:
"Function is self describing no need for documentation"

...

I'll add them.

---
In reply to: [591987004](https://github.com/Azure/iot-hub-device-update/pull/12#discussion_r591987004) [](ancestors = 591987004,591977401)